src/pages/Swap.tsx

```diff
@@ -131,21 +131,26 @@ function Swap(): ReactElement {
       .filter(({ isLPToken }) => !isLPToken)
       .filter(({ symbol }) => {
         // get list of pools containing the token
-        const tokenPools = tokenToPoolsMap[symbol] ?? []
+        if (!tokenSymbolToPoolNameMap[symbol]) return false
+        const tokenPools = tokenSymbolToPoolNameMap[symbol]
         // ensure at least one pool is unpaused to include token in swappable list
         const hasAnyUnpaused = tokenPools.some((poolName) => {
-          const basicPool = basicPools?.[poolName]
-          return basicPool ? !basicPool.isPaused : false
+          if (!basicPools) return false
+          const basicPool = basicPools[poolName]
+          if (!basicPool) return false
+          return !basicPool.isPaused
         })
         // only show pools with balances
         const hasAnyBalance = tokenPools.some((poolName) => {
+          if (!basicPools) return false
           const basicPool = basicPools?.[poolName]
-          return basicPool?.lpTokenSupply.gt(Zero) ?? false
+          if (!basicPool) return false
+          return basicPool.lpTokenSupply.gt(Zero)
```

Comment:
Up to your discretion, but this is repeated more than once, so breaking out into a helper function could be beneficial here.

Reply:
agree though I think we should revisit this in the future with a refactor to simplify some logic

Reply:
Got it. I'll add all of these issues into a big `FrontEnd Clean up` epic.  Our job right now is to reduce as many extraneous optionals as possible while keeping things operational.